### PR TITLE
Refactor subroutine which creates xml result file

### PR DESCRIPTION
**Pull request by b10n1k**
Move the parsing and the creation of the xml out of the SUT. As such we do not repeat the script_run script which might make the runtime a bit faster overall.
In addition I added date and build elements in the xml as additional info of the test results which i think will help in the review of the tests.

- Related ticket: https://progress.opensuse.org/issues/127673
- Verification run: https://openqa.suse.de/tests/overview?version=15-SP6&build=b10n1k%2Fos-autoinst-distri-opensuse%2318021&distri=sle

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -2627,54 +2628,50 @@ sub generate_results {
...
+    create_xml_results();
 
 Takes C<test> as an argument. C<test> is an array of hashes which contain the
 test results. They usually are generated by C<generate_results>. Those are
 parsed and create the junit xml representation.
 
 =cut
 
-sub pars_results {
+sub create_xml_results {
```
[the remark this answers is not in the document]
Comment (b10n1k) [reply lib/utils.pm:2640]: obviously the pars_result should be parse_results. But I thought to change it as part of this PR and considering what exactly the function is doing. i couldnt think anything else as well. But i think it is very accurate. i wonder what do you think it should be based on its functionality

**Review by schlad — CHANGES_REQUESTED** [review]
Stop writing such stuff in the git commit messages: "which i think will help in the review of the tests."
Not to mention one does not write "i" but "I".
In any case as discussed and pointed before: git commit message should not be written in a personal form "I did/I think etc."

File: lib/utils.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (schlad) [reply lib/utils.pm:2640]: ultimately this is the main thing:
if ($i->{result} eq 'FAIL') {
            $tc_elem->setAttribute(error => '1');

so this is the test result parser. I would prefer you change the name to the pars_test_results

**Comment by b10n1k** [comment]
> Stop writing such stuff in the git commit messages: "which i think will help in the review of the tests." Not to mention one does not write "i" but "I". In any case as discussed and pointed before: git commit message should not be written in a personal form "I did/I think etc."

Updated

File: lib/utils.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (b10n1k) [reply lib/utils.pm:2640]: `pars_test_results` or `parse_test_results`?

**Review by schlad — APPROVED** [review]

**Pull request merged**